nautobot/core/settings.py

```diff
@@ -805,3 +805,35 @@ def UI_RACK_VIEW_TRUNCATE_FUNCTION(device_display_name):
     :type: str
     """
     return str(device_display_name).split(".")[0]
+
+
+# Custom JSON schema serializer field type mappingss. These will be added to
+# `NautobotProcessingMixin.TYPE_MAP`.
+# Format: `{serializer_field_class.__name__}` => `{json_schema_type}`
+# See: https://github.com/yoyowallet/drf-react-template-framework#settings
+DRF_REACT_TEMPLATE_TYPE_MAP = {
+    "ContentTypeField": {"type": "string", "enum": "choices"},
+    "CustomFieldsDataField": {"type": "object", "widget": "textarea"},
+    "DateTimeField": {"type": "string", "format": "date-time", "widget": "date-time"},
+    "ImageField": {"type": "string", "format": "data-url"},
+    "IPFieldSerializer": {"type": "string"},
+    "JSONField": {"type": "string", "widget": "textarea"},
+    "MultipleChoiceJSONField": {"type": "array", "required": [], "enum": "choices"},
+    "ManyRelatedField": {"type": "array", "required": []},
+    #
+    # Foreign Key fields
+    #
+    # enum=choices is the one that works in the UI as a related field but it
+    # includes ALL related objects in the schema.
+    # "NautobotHyperlinkedRelatedField": {"type": "string", "enum": "choices"},
+    # readOnly=True disables the fields in the UI; not what we want.
+    # "NautobotHyperlinkedRelatedField": {"type": "string", "readOnly": True},
+    # type=string results in a free text field; also not what we want. For now,
+    # however, this will keep things moving so the unit tests pass.
+    "NautobotHyperlinkedRelatedField": {"type": "string", "format": "uuid"},
```

Comment:
TODO: Should this be `"format": "url"`?

Reply:
Looks like it would be [`uri`](https://rjsf-team.github.io/react-jsonschema-form/docs/usage/widgets/#string-formats) but we should probably hold off on making any changes that assume we're going to keep using rjsf

Reply:
It can be a UUID on write.